src/intervals/arithmetic/basic.jl

```diff
@@ -0,0 +1,302 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+#=  This file contains the functions described as "Basic arithmetic functions"
+    in the IEEE Std 1788-2015 (sections 9.1) and required for set-based flavor
+    in section 10.5.3, at the exception of the `sqr` function present in the
+    "Power functions" file.
+=#
+
++(a::Interval) = a  # Not in the IEEE standard
+
+"""
+    -(a::Interval)
+
+Implement the `neg` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+-(a::F) where {F<:Interval} = F(-a.hi, -a.lo)
+
+
+"""
+    +(a::Interval, b::Real)
+    +(a::Real, a::Interval)
+    +(a::Interval, b::Interval)
+
+Implement the `add` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function +(a::F, b::T) where {T, F<:Interval{T}}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, a.lo + b, a.hi + b)
+end
++(a::Interval{T}, b::S) where {T, S<:Real} = a + Interval{T}(b)
++(b::Real, a::Interval) = a + b
+
+function +(a::F, b::F) where {F<:Interval}
+    (isempty(a) || isempty(b)) && return emptyinterval(F)
+    return @round(F, a.lo + b.lo, a.hi + b.hi)
+end
+
+"""
+    -(a::Interval, b::Real)
+    -(a::Real, a::Interval)
+    -(a::Interval, b::Interval)
+
+Implement the `sub` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function -(a::F, b::Real) where {F<:Interval}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, a.lo - b, a.hi - b)
+end
+
+function -(b::Real, a::F) where {F<:Interval}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, b - a.hi, b - a.lo)
+end
+
+function -(a::F, b::F) where {T<:Real, F<:Interval{T}}
+    (isempty(a) || isempty(b)) && return emptyinterval(F)
+    return @round(F, a.lo - b.hi, a.hi - b.lo)
+end
+
+-(a::Interval{T}, b::S) where {T<:Real, S<:Real} = a - Interval{T}(b)
+-(a::T, b::Interval{S}) where {T<:Real, S<:Real} = Interval{S}(a) - b
```

Comment:
Naive question: aren't these methods included already in those of lines 45 and 50?

Reply:
Those were a very weird way to rdefine fallback for when the type of the bound does not match the type of the real being substracted.

The additional methods are needed so that Irrational number are correctly rounded (something I added in this PR) e.g. without it `Interval(π) - (0..0)` is different from `Interval(π)`.

I have changed it to make it clearer, requiring explicitly the same type in the specific version.